[iOS] Remove all instances of UICalloutBar in WebKit

whsieh opened the pull request:
#### 127b86b20b91c47794e615285dde394a221ae6c3
<pre>
[iOS] Remove all instances of UICalloutBar in WebKit
<a href="https://bugs.webkit.org/show_bug.cgi?id=259197">https://bugs.webkit.org/show_bug.cgi?id=259197</a>
rdar://112114237

Reviewed by Megan Gardner and Aditya Keerthi.

`UICalloutBar` has been long replaced by the edit menu (and its associated interaction) since iOS
16. This patch removes several codepaths that only existed to support layout and API tests against
iOS 15 — this is no longer necessary, now that our oldest supported iOS test runners are on 16.4.

No change in behavior.

* Source/WebKit/Platform/spi/ios/UIKitSPI.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView _requestEvasionRectsAboveSelectionIfNeeded:]):

Additionally rename several references to &quot;callout bar&quot; to &quot;edit menu&quot; instead.

(-[WKContentView cleanUpDragSourceSessionState]):
(-[WKContentView _restoreEditMenuIfNeeded]):
(-[WKContentView dragInteraction:willAnimateLiftWithAnimator:session:]):
(-[WKContentView dragInteraction:session:didEndWithOperation:]):
(-[WKContentView _restoreCalloutBarIfNeeded]): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ImageAnalysisTests.mm:
(TestWebKitAPI::simulateEditMenuAppearance):
(TestWebKitAPI::invokeRemoveBackgroundAction):
(TestWebKitAPI::TEST):

Remove an obsolete test that was only checking that we avoid eagerly instantiating `UICalloutBar`
when changing the text selection. This test is no longer relevant with edit menus because the bug
was triggered by callout-bar-specific code that was invoked by WebKit code; in comparison, edit
menus ask the first responder (the content view) to build the edit menu upon presentation.

(TestWebKitAPI::simulateCalloutBarAppearance): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKAppHighlights.mm:
* Tools/TestWebKitAPI/cocoa/TestWKWebView.mm:
(-[TestWKWebView initWithFrame:configuration:addToWindow:]):
(suppressUICalloutBar): Deleted.
* Tools/TestWebKitAPI/ios/UIKitSPI.h:
* Tools/WebKitTestRunner/ios/UIScriptControllerIOS.mm:
(WTR::UIScriptControllerIOS::rectForMenuAction const):
(WTR::UIScriptControllerIOS::menuRect const):

Canonical link: <a href="https://commits.webkit.org/266047@main">https://commits.webkit.org/266047@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/09497ddd4531e05ac5a30c6224b6a887fb035524

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/12663 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/12991 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/13315 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/14402 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/12121 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/12724 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/15499 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/13009 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/14820 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/12827 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/13598 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/10720 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/14845 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/10876 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/11466 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/18546 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/11961 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/11632 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/14829 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/12107 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/10021 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/11355 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3110 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/15690 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/11964 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas reviewed (approved):
lgtm

megangardner commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/WKAppHighlights.mm:
```diff
@@ -215,21 +215,6 @@ - (void)_webView:(WKWebView *)webView storeAppHighlight:(_WKAppHighlight *)highl
     }, 2, @"Expected Highlights to be populated.");
 }
 
-#if PLATFORM(IOS_FAMILY)
```
Do we need to replace this test with an edit menu equivalent?

whsieh replied:
No (due to the reasons in the commit message)

whsieh committed:
[iOS] Remove all instances of UICalloutBar in WebKit
https://bugs.webkit.org/show_bug.cgi?id=259197
rdar://112114237

Reviewed by Megan Gardner and Aditya Keerthi.

`UICalloutBar` has been long replaced by the edit menu (and its associated interaction) since iOS
16. This patch removes several codepaths that only existed to support layout and API tests against
iOS 15 — this is no longer necessary, now that our oldest supported iOS test runners are on 16.4.

No change in behavior.

* Source/WebKit/Platform/spi/ios/UIKitSPI.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView _requestEvasionRectsAboveSelectionIfNeeded:]):

Additionally rename several references to "callout bar" to "edit menu" instead.

(-[WKContentView cleanUpDragSourceSessionState]):
(-[WKContentView _restoreEditMenuIfNeeded]):
(-[WKContentView dragInteraction:willAnimateLiftWithAnimator:session:]):
(-[WKContentView dragInteraction:session:didEndWithOperation:]):
(-[WKContentView _restoreCalloutBarIfNeeded]): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ImageAnalysisTests.mm:
(TestWebKitAPI::simulateEditMenuAppearance):
(TestWebKitAPI::invokeRemoveBackgroundAction):
(TestWebKitAPI::TEST):

Remove an obsolete test that was only checking that we avoid eagerly instantiating `UICalloutBar`
when changing the text selection. This test is no longer relevant with edit menus because the bug
was triggered by callout-bar-specific code that was invoked by WebKit code; in comparison, edit
menus ask the first responder (the content view) to build the edit menu upon presentation.

(TestWebKitAPI::simulateCalloutBarAppearance): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKAppHighlights.mm:
* Tools/TestWebKitAPI/cocoa/TestWKWebView.mm:
(-[TestWKWebView initWithFrame:configuration:addToWindow:]):
(suppressUICalloutBar): Deleted.
* Tools/TestWebKitAPI/ios/UIKitSPI.h:
* Tools/WebKitTestRunner/ios/UIScriptControllerIOS.mm:
(WTR::UIScriptControllerIOS::rectForMenuAction const):
(WTR::UIScriptControllerIOS::menuRect const):

Canonical link: https://commits.webkit.org/266047@main
